bin/varnishd/cache/cache_vrt_var.c

```diff
@@ -627,7 +629,11 @@ VRT_r_##which##_##fld(VRT_CTX)					\
 
 VRT_DO_EXP_R(obj, ctx->req->objcore, ttl,
     ttl_now(ctx) - ctx->req->objcore->t_origin)
-VRT_DO_EXP_R(obj, ctx->req->objcore, grace, 0)
+VRT_DO_EXP_R(obj, ctx->req->objcore, grace,
+    ctx->req->d_grace >= 0
+    && ctx->req->d_grace < ctx->req->objcore->grace
+    ? ctx->req->objcore->grace - ctx->req->d_grace
+    : 0)
```

Comment:
Two questions:

1. Should obj.grace really take req.grace into account ?

2. If yes, why not call EXP_Ttl_grace() ?